ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/PrivJsonRpcMethods.java

```diff
@@ -65,57 +66,44 @@ protected RpcApi getApiGroup() {
-              privacyController,
-              enclavePublicKeyProvider),
-          new PrivGetPrivacyPrecompileAddress(getPrivacyParameters()),
-          new PrivGetTransactionCount(privacyController, enclavePublicKeyProvider),
-          new PrivGetPrivateTransaction(privacyController, enclavePublicKeyProvider),
-          new PrivDistributeRawTransaction(
-              privacyController,
-              enclavePublicKeyProvider,
-              getPrivacyParameters().isOnchainPrivacyGroupsEnabled()),
-          new PrivCall(getBlockchainQueries(), privacyController, enclavePublicKeyProvider),
-          new PrivGetCode(getBlockchainQueries(), privacyController, enclavePublicKeyProvider),
-          new PrivGetLogs(
-              getBlockchainQueries(),
-              getPrivacyQueries(),
-              privacyController,
-              enclavePublicKeyProvider),
-          new PrivNewFilter(filterManager, privacyController, enclavePublicKeyProvider),
-          new PrivUninstallFilter(filterManager, privacyController, enclavePublicKeyProvider),
-          new PrivGetFilterLogs(filterManager, privacyController, enclavePublicKeyProvider),
-          new PrivGetFilterChanges(filterManager, privacyController, enclavePublicKeyProvider));
-    } else {
-      return mapOf(
-          new PrivGetTransactionReceipt(
-              getPrivacyParameters().getPrivateStateStorage(),
-              privacyController,
-              enclavePublicKeyProvider),
-          new PrivCreatePrivacyGroup(privacyController, enclavePublicKeyProvider),
-          new PrivDeletePrivacyGroup(privacyController, enclavePublicKeyProvider),
-          new PrivFindPrivacyGroup(privacyController, enclavePublicKeyProvider),
-          new PrivGetPrivacyPrecompileAddress(getPrivacyParameters()),
-          new PrivGetTransactionCount(privacyController, enclavePublicKeyProvider),
-          new PrivGetPrivateTransaction(privacyController, enclavePublicKeyProvider),
-          new PrivDistributeRawTransaction(
-              privacyController,
-              enclavePublicKeyProvider,
-              getPrivacyParameters().isOnchainPrivacyGroupsEnabled()),
-          new PrivCall(getBlockchainQueries(), privacyController, enclavePublicKeyProvider),
-          new PrivGetCode(getBlockchainQueries(), privacyController, enclavePublicKeyProvider),
-          new PrivGetLogs(
-              getBlockchainQueries(),
-              getPrivacyQueries(),
-              privacyController,
-              enclavePublicKeyProvider),
-          new PrivNewFilter(filterManager, privacyController, enclavePublicKeyProvider),
-          new PrivUninstallFilter(filterManager, privacyController, enclavePublicKeyProvider),
-          new PrivGetFilterLogs(filterManager, privacyController, enclavePublicKeyProvider),
-          new PrivGetFilterChanges(filterManager, privacyController, enclavePublicKeyProvider));
+
+    final Map<String, JsonRpcMethod> RPC_METHODS =
```

Comment:
I am happy to see this refactor

Reply:
adding a unit test would take it to next level

Reply:
The existing tests have good coverage of what I've changed. 👍

Reply:
even better